manipulation/src/main/java/org/jboss/gm/manipulation/actions/PublishingArtifactsAction.java

```diff
@@ -28,22 +29,27 @@
 public class PublishingArtifactsAction implements Action<Project> {
     @Override
     public void execute(Project project) {
-        project.getExtensions().configure(PublishingExtension.class, new Action<PublishingExtension>() {
-            @Override
-            public void execute(PublishingExtension publishingExtension) {
-                publishingExtension.publications(new Action<PublicationContainer>() {
-                    @Override
-                    public void execute(PublicationContainer publications) {
-                        publications.create("mavenJava", MavenPublication.class, new Action<MavenPublication>() {
-                            @Override
-                            public void execute(MavenPublication mavenPublication) {
-                                mavenPublication.from(project.getComponents().getByName("java"));
-                                // TODO: why are optional dependencies missing in generated pom.xml?
-                            }
-                        });
-                    }
-                });
-            }
-        });
+        SoftwareComponent javaComponent = project.getComponents().findByName("java");
+        if (javaComponent != null) {
+            project.getExtensions().configure(PublishingExtension.class, new Action<PublishingExtension>() {
+                @Override
+                public void execute(PublishingExtension publishingExtension) {
+                    publishingExtension.publications(new Action<PublicationContainer>() {
+                        @Override
+                        public void execute(PublicationContainer publications) {
+                            publications.create("mavenJava", MavenPublication.class, new Action<MavenPublication>() {
+                                @Override
+                                public void execute(MavenPublication mavenPublication) {
+                                    mavenPublication.from(javaComponent);
+                                }
+                            });
+                        }
+                    });
+                }
+            });
+        } else {
+            project.getLogger().warn("No java component found for project {}, no artifact to publish.",
+                    project.getName());
```

Comment:
Does this mean we don't publish then ? i.e. the build will complete but nothing will get uploaded?

Reply:
For given module, yes. I think root module would be often time the case.

Reply:
@TomasHofman Sorry didn't quite understand your last comment? I suppose the problem is at this point we know the build might fail to deploy, but failing in the alignment stage doesn't seem right either.

Potentially switch to error logging to try and highlight it better?

Also please note https://github.com/project-ncl/gradle-manipulator/blob/master/CONTRIBUTING.md#code-recommendations - any logging **must** use GMLogger now.

Reply:
Discussed offline and agreed to just change logger and nothing else